src/lib.rs

```diff
@@ -28,23 +28,57 @@ use errors::Error;
 use hashlock::{sha256, Preimage};
 
 use bitcoin::{
-    sighash::Prevouts, transaction::InputWeightPrediction, OutPoint, Transaction, TxIn, TxOut,
+    secp256k1::XOnlyPublicKey as BitcoinXOnly, sighash::Prevouts,
+    transaction::InputWeightPrediction, OutPoint, Transaction, TxIn, TxOut,
+};
+use musig2::{
+    secp256k1::XOnlyPublicKey as Musig2XOnly, AdaptorSignature, AggNonce, CompactSignature,
+    PartialSignature, PubNonce, SecNonce,
 };
-use musig2::{AdaptorSignature, AggNonce, CompactSignature, PartialSignature, PubNonce, SecNonce};
 use secp::{MaybeScalar, Point, Scalar};
 use serde::{Deserialize, Serialize};
 
-use std::{
-    borrow::Borrow,
-    collections::{BTreeMap, BTreeSet},
-};
-
 pub use contract::{
     ContractParameters, Outcome, OutcomeIndex, PayoutWeights, PlayerIndex, SigMap, WinCondition,
 };
 pub use oracles::{attestation_locking_point, attestation_secret, EventLockingConditions};
 pub use parties::{MarketMaker, Player};
 
+use std::{
+    borrow::Borrow,
+    collections::{BTreeMap, BTreeSet},
+};
+
+/// Used to convert musig2 xonly key into bitcoin xonly key only needed
+/// until crate bitcoin updates secp256k1 to v0.30
+/// will be removed/deprecated once the dependency has upgrade
+pub fn convert_xonly_key(key: Musig2XOnly) -> BitcoinXOnly {
+    let serialized = key.serialize();
+
+    // Create a fixed size array for the x-coordinate
+    let x_bytes: [u8; 32] = if serialized.len() == 32 {
+        serialized.try_into().expect("Length already checked")
+    } else if serialized.len() == 33 {
+        serialized[1..]
+            .try_into()
+            .expect("Slice to array conversion failed")
+    } else {
+        panic!("Unexpected key length: {}", serialized.len())
+    };
+
+    BitcoinXOnly::from_slice(&x_bytes).expect("Valid key")
```

Comment:
It's impossible for [the `XOnlyPublicKey::serialize()` method](https://docs.rs/secp256k1/0.30.0/secp256k1/struct.XOnlyPublicKey.html#method.serialize) to return an array with any length other than 32, so can you please simplify here?

This should be perfectly safe.

```suggestion
    BitcoinXOnly::from_slice(&key.serialize()).expect("Valid key")
```